pkg/apis/extensions/latest/latest.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package latest
```

Comment:
> The point is to enable code to be able to iterate over all group/versions that are compiled in/registered.

I definitely agree.  Now we know that there are only two places in our code that handle this correctly and we can start fixing them.

This pull doesn't change any existing behavior.  It's just making it obvious that the existing code doesn't do what you want it to.